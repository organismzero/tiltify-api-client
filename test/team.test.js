const TiltifyClient = require('../index')

describe('teams tests -', () => {
  beforeAll(() => {
    this.client = new TiltifyClient('','').setKey(process.env.TILTIFY_ACCESS_TOKEN)
  })

  test('correct Tiltify team returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot({
        total_amount_raised: expect.any(Object)
      })
      done()
    }
    this.client.Team.get('459', callback)
  })

  test('correct Tiltify team campaigns returned', done => {
    function callback (data) {
      data.forEach(element => {
        expect(element).toMatchSnapshot({
          total_amount_raised: expect.any(Object),
          supporting_amount_raised: expect.any(Object)
        })
      });
      done()
    }
    this.client.Team.getCampaigns('459', callback)
  })

})
