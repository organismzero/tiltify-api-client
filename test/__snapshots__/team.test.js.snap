// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teams tests - correct Tiltify team campaigns returned 1`] = `
Object {
  "amount_raised": Object {
    "currency": "USD",
    "value": "225.00",
  },
  "avatar": Object {
    "alt": "alt",
    "height": 64,
    "src": "https://assets.tiltify.com/uploads/team_event/avatar/d37d437c-a9f4-4d54-b064-3715863c00c3/Z.jpg",
    "width": 64,
  },
  "cause_id": "400f5687-6017-4d1a-a4d9-7c9166b984c2",
  "currency_code": "USD",
  "description": "Play video games and create a fundraising campaign to help kids battling cancer.",
  "fundraising_event_id": "58faabc0-e4e3-4085-b6ed-0180c265ace8",
  "goal": Object {
    "currency": "USD",
    "value": "5000.0",
  },
  "has_schedule": false,
  "id": "d37d437c-a9f4-4d54-b064-3715863c00c3",
  "inserted_at": "2018-04-02T20:34:03Z",
  "legacy_id": 13067,
  "livestream": Object {
    "channel": "Lizerdoo",
    "type": "twitch",
  },
  "name": "Zeldathon Community PLAYLIVE 2018",
  "original_goal": Object {
    "currency": "USD",
    "value": "5000.0",
  },
  "published_at": "2018-05-13T22:52:59Z",
  "retired_at": "2018-05-13T22:52:59Z",
  "slug": "zeldathon-community-playlive-2018",
  "status": "retired",
  "supportable": "public",
  "supporting_amount_raised": Any<Object>,
  "team": Object {
    "avatar": Object {
      "alt": "alt",
      "height": 200,
      "src": "https://assets.tiltify.com/uploads/team/thumbnail/459/Z.jpg",
      "width": 200,
    },
    "description": "Zeldathon loves St Jude, and has raised over $500,000 with our marathons supporting them so far. Join us as we venture forth!
For the kids!",
    "id": "32ddfc95-45a8-47b9-b003-8028224a8ec1",
    "legacy_id": 459,
    "name": "Zeldathon Community",
    "slug": "zeldathon-community",
    "social": Object {
      "discord": null,
      "facebook": "Zeldathon",
      "instagram": null,
      "snapchat": null,
      "tiktok": null,
      "twitch": "SuperMCGamer",
      "twitter": "@Zeldathons",
      "website": "zeldathon.net",
      "youtube": "SuperMCGamer",
    },
    "total_amount_raised": Object {
      "currency": "USD",
      "value": "2787.00",
    },
    "url": "https://tiltify.com/+zeldathon-community",
  },
  "team_id": "32ddfc95-45a8-47b9-b003-8028224a8ec1",
  "total_amount_raised": Any<Object>,
  "updated_at": "2022-06-14T17:23:41Z",
  "url": "https://tiltify.com/+zeldathon-community/zeldathon-community-playlive-2018",
  "user": null,
  "user_id": null,
}
`;

exports[`teams tests - correct Tiltify team campaigns returned 2`] = `
Object {
  "amount_raised": Object {
    "currency": "USD",
    "value": "0.00",
  },
  "avatar": Object {
    "alt": "alt",
    "height": 64,
    "src": "https://assets.tiltify.com/uploads/team_event/avatar/4c356d48-005f-43c1-8b9d-e313c53de354/df382276-0ee0-4687-b09d-ea22ed1939f1.jpeg",
    "width": 64,
  },
  "cause_id": "400f5687-6017-4d1a-a4d9-7c9166b984c2",
  "currency_code": "USD",
  "description": "St. Jude PLAY LIVE is where gamers can help the kids of St. Jude Childrenâ€™s Research Hospital by doing what they love-playing video games. Sign-up, set a goal, earn awesome prizes today!",
  "fundraising_event_id": "663d89f6-a7b0-47e6-b9c1-a4eeeb01a615",
  "goal": Object {
    "currency": "USD",
    "value": "5000.0",
  },
  "has_schedule": false,
  "id": "4c356d48-005f-43c1-8b9d-e313c53de354",
  "inserted_at": "2019-04-29T16:48:44Z",
  "legacy_id": 27490,
  "livestream": Object {
    "channel": "lizerdoo",
    "type": "twitch",
  },
  "name": "ZthonPlayLive2019",
  "original_goal": Object {
    "currency": "USD",
    "value": "5000.0",
  },
  "published_at": "2019-04-29T16:56:25Z",
  "retired_at": "2019-04-29T16:56:25Z",
  "slug": "zthonplaylive2019",
  "status": "retired",
  "supportable": "public",
  "supporting_amount_raised": Any<Object>,
  "team": Object {
    "avatar": Object {
      "alt": "alt",
      "height": 200,
      "src": "https://assets.tiltify.com/uploads/team/thumbnail/459/Z.jpg",
      "width": 200,
    },
    "description": "Zeldathon loves St Jude, and has raised over $500,000 with our marathons supporting them so far. Join us as we venture forth!
For the kids!",
    "id": "32ddfc95-45a8-47b9-b003-8028224a8ec1",
    "legacy_id": 459,
    "name": "Zeldathon Community",
    "slug": "zeldathon-community",
    "social": Object {
      "discord": null,
      "facebook": "Zeldathon",
      "instagram": null,
      "snapchat": null,
      "tiktok": null,
      "twitch": "SuperMCGamer",
      "twitter": "@Zeldathons",
      "website": "zeldathon.net",
      "youtube": "SuperMCGamer",
    },
    "total_amount_raised": Object {
      "currency": "USD",
      "value": "2787.00",
    },
    "url": "https://tiltify.com/+zeldathon-community",
  },
  "team_id": "32ddfc95-45a8-47b9-b003-8028224a8ec1",
  "total_amount_raised": Any<Object>,
  "updated_at": "2022-06-14T17:21:07Z",
  "url": "https://tiltify.com/+zeldathon-community/zthonplaylive2019",
  "user": null,
  "user_id": null,
}
`;

exports[`teams tests - correct Tiltify team returned 1`] = `
Object {
  "about": "Zeldathon loves St Jude, and has raised over $500,000 with our marathons supporting them so far. Join us as we venture forth!
For the kids!",
  "avatar": Object {
    "alt": "",
    "height": 200,
    "src": "https://assets.tiltify.com/uploads/team/thumbnail/459/thumb_Z.jpg",
    "width": 200,
  },
  "id": 459,
  "inviteOnly": false,
  "name": "Zeldathon Community",
  "slug": "zeldathon-community",
  "social": Object {
    "discord": null,
    "facebook": "Zeldathon",
    "mixer": null,
    "twitch": "SuperMCGamer",
    "twitter": "@Zeldathons",
    "website": "zeldathon.net",
    "youtube": "SuperMCGamer",
  },
  "total_amount_raised": Any<Number>,
  "url": "https://tiltify.com/+zeldathon-community",
}
`;
